RInAppMessaging/Classes/Repositories/IAMPreferenceRepository.swift

```diff
@@ -1,11 +1,21 @@
 /// Repository to hold `IAMPreference`.
 internal class IAMPreferenceRepository {
-    private(set) var preference: IAMPreference?
+    private(set) var preference: UserInfoProvider?
 
-    func setPreference(_ preference: IAMPreference?) {
+    func setPreference(_ preference: UserInfoProvider?) {
         self.preference = preference
     }
 
+    func canUpdateUserInfo(newUserInfo: UserInfoProvider?) -> Bool {
```

Comment:
Maybe this signature is more swifty:
`func canUpdate(userInfo newUserInfo: UserInfoProvider?) -> Bool`